src/CoreRegistry.cpp

```diff
@@ -42,37 +42,38 @@ void CoreRegistry::start(HttpServer *svr)
 
     // initialize input plugins
     {
-        auto alias_list = _input_registry.aliasList();
         auto plugin_list = _input_registry.pluginList();
-        std::vector<std::string> by_alias;
-        std::set_difference(alias_list.begin(), alias_list.end(),
-            plugin_list.begin(), plugin_list.end(), std::inserter(by_alias, by_alias.begin()));
-        for (auto &s : by_alias) {
+        for (auto &s : plugin_list) {
             auto meta = _input_registry.metadata(s);
             if (!meta) {
                 _logger->error("failed to load plugin metadata: {}", s);
                 continue;
             }
             if (meta->data().hasValue("type") && meta->data().value("type") == "input") {
+                if (!meta->data().hasValue("version")) {
+                    _logger->error("version field is mandatory and was not provided by '{}'", s);
+                }
+                auto version = meta->data().value("version");
                 if (_input_registry.loadState(s) == Corrade::PluginManager::LoadState::NotLoaded) {
                     _input_registry.load(s);
                 }
-                InputPluginPtr mod = _input_registry.instantiate(s);
-                _logger->info("Load input stream plugin: {} {}", s, mod->pluginInterface());
-                mod->init_plugin(this, svr);
-                _input_plugins.insert({s, std::move(mod)});
+                for (const auto &alias : meta->provides()) {
+                    InputPluginPtr mod = _input_registry.instantiate(alias);
+                    _logger->info("Load input stream plugin: {} {}", alias, mod->pluginInterface());
```

Comment:
let's add the version to this message too

Reply:
done